gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpRequestRunnable.java

```diff
@@ -120,6 +121,21 @@ HttpRequest createHttpRequest() throws IOException {
     return httpRequest;
   }
 
+  // This will be frequently executed, so avoiding using regexps if not necessary.
+  private String normalizeEndpoint(String endpoint) {
```

Comment:
Optional: Consider adding a unit test and making this package-private under `@VisibleForTesting` so we can see the expected input values. Alternatively, add a comment with example expected input values.

Reply:
It turned out that it was easy to write a legitimate test for this functionality as part of `HttpRequestrunnableTest` without need to rise the visibility level of the method. Thanks for pointing out this missing test coverage!